sphinx/builders/html/__init__.py

```diff
@@ -199,7 +200,7 @@ def __init__(self, app: Sphinx, env: BuildEnvironment) -> None:
         self._js_files: list[_JavaScript] = []
 
         # Cached Publisher for writing doctrees to HTML
-        reader = docutils.readers.doctree.Reader(parser_name='restructuredtext')
+        reader: Reader = docutils.readers.doctree.Reader(parser_name='restructuredtext')
```

Comment:
actually i think it's because it's a generic class and the 'source' type is not specified:

```python
class Reader(Component, Generic[_S]):
    component_type: ClassVar[str]
    config_section: ClassVar[str]
    def get_transforms(self) -> list[type[Transform]]: ...
    def __init__(self, parser: Parser | None = None, parser_name: str | None = None) -> None: ...
    parser: Parser | None
    source: Input[_S] | None
    input: str | Any | None
    def set_parser(self, parser_name: str) -> None: ...
    def read(self, source: Input[_S], parser: Parser, settings: Values) -> nodes.document: ...
    document: nodes.document
    def parse(self) -> None: ...
    def new_document(self) -> nodes.document: ...
```